src/libraries/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -196,7 +196,7 @@ private int InternalGetHashCode(TKey key)
         internal Grouping<TKey, TElement>? GetGrouping(TKey key, bool create)
         {
             int hashCode = InternalGetHashCode(key);
-            for (Grouping<TKey, TElement>? g = _groupings[hashCode % _groupings.Length]; g != null; g = g._hashNext)
+            for (Grouping<TKey, TElement>? g = _groupings[(uint)hashCode % _groupings.Length]; g != null; g = g._hashNext)
```

Comment:
This doesn't also need _groupings stored into a local?

Reply:
> This doesn't also need _groupings stored into a local?

Yes, but I wasn't sure I could hoist it since in this method it's used as field in all places, presumably, the implementation is not thread safe?

My change here just converts signed MOD to unsigned (slightly better codegen)